script.js

```diff
@@ -1,4 +1,340 @@
+// Variables to store player 1 and player 2's scores:
+var P1TotalScore = 0;
+var P2TotalScore = 0;
+
+// To track game modes & statuses
+var MODE_CHOOSE_GAME_MODE = `WAITING_FOR_USER_TO_CHOOSE_GAME_MODE`;
+var EITHER_HIGHEST_OR_LOWEST_WINS = `EITHER HIGHEST OR LOWEST WINS`;
+var MODE_HIGHER_NUMBER = `higher number`;
+var MODE_LOWER_NUMBER = `lower number`;
+var MODE_PLAYER_1_ROLL_DICE = `WAITING_FOR_PLAYER_1_TO_ROLL_DICE`;
+var MODE_PLAYER_1_PICK_ORDER = `WAITING_FOR_PLAYER_1_TO_PICK_DICE_ORDER`;
+var MODE_PLAYER_2_ROLL_DICE = `WAITING_FOR_PLAYER_2_TO_ROLL_DICE`;
+var MODE_PLAYER_2_PICK_ORDER = `WAITING_FOR_PLAYER_2_TO_PICK_DICE_ORDER`;
+var MODE_WINNER = `SHOW_WINNER`;
+
+//Current game mode
+var mode = MODE_CHOOSE_GAME_MODE;
+var theModeUserInput = EITHER_HIGHEST_OR_LOWEST_WINS;
+
+// Function for leaderboard for Mode: Higher number wins:
+var leaderBoardHigher = function () {
+  if (Number(P1TotalScore) > Number(P2TotalScore)) {
+    myOutputValue = `Player 1<br>Player 2`;
+  } else if (Number(P2TotalScore) > Number(P1TotalScore)) {
+    myOutputValue = `Player 2<br>Player 1`;
+  } else if (Number(P1TotalScore) == Number(P2TotalScore)) {
+    myOutputValue = `Draw`;
+  }
+  return myOutputValue;
+};
+
+// Function for leaderboard for Mode: Lower number wins:
+var leaderBoardLower = function () {
```

Comment:
We could try having a more semantic name for `leaderBoardLower` .  Additionally, in general, a good practice is to use  a verb as the first word in function names. like  **calculate**TotalSum, or **get**AllDice.